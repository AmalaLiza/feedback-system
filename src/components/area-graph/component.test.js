import React from "react";
import { shallow } from "enzyme";
import AreaGraph from "./component";
import renderer from "react-test-renderer";

const props = {
	chartData: [
		{
			rating: 5,
			count: 2,
		},
		{ rating: 5, count: 2 },
	],
	xAxisDataKey: "rating",
	areaDataKey: "count",
};

function setup() {
	const wrapper = shallow(<AreaGraph {...props} />);
	return { wrapper, props };
}

describe("AreaGraph Component Test Suite", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<AreaGraph {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should have an AreaChart", () => {
		const { wrapper } = setup();
		expect(wrapper.find("AreaChart").exists()).toBe(true);
	});
});
