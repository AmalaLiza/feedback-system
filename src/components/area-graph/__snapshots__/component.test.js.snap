// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaGraph Component Test Suite renders correctly 1`] = `
<div
  className="recharts-wrapper"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "cursor": "default",
      "height": 400,
      "position": "relative",
      "width": 500,
    }
  }
>
  <svg
    className="recharts-surface"
    fillOpacity={1}
    height={400}
    version="1.1"
    viewBox="0 0 500 400"
    width={500}
  >
    <defs>
      <clipPath
        id="recharts2-clip"
      >
        <rect
          height={360}
          width={410}
          x={60}
          y={10}
        />
      </clipPath>
    </defs>
    <g
      className="recharts-cartesian-grid"
    >
      <g
        className="recharts-cartesian-grid-horizontal"
      >
        <line
          fill="none"
          height={360}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={410}
          x={60}
          x1={60}
          x2={470}
          y={10}
          y1={370}
          y2={370}
        />
        <line
          fill="none"
          height={360}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={410}
          x={60}
          x1={60}
          x2={470}
          y={10}
          y1={298}
          y2={298}
        />
        <line
          fill="none"
          height={360}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={410}
          x={60}
          x1={60}
          x2={470}
          y={10}
          y1={226}
          y2={226}
        />
        <line
          fill="none"
          height={360}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={410}
          x={60}
          x1={60}
          x2={470}
          y={10}
          y1={154}
          y2={154}
        />
        <line
          fill="none"
          height={360}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={410}
          x={60}
          x1={60}
          x2={470}
          y={10}
          y1={81.99999999999999}
          y2={81.99999999999999}
        />
        <line
          fill="none"
          height={360}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={410}
          x={60}
          x1={60}
          x2={470}
          y={10}
          y1={10}
          y2={10}
        />
      </g>
      <g
        className="recharts-cartesian-grid-vertical"
      >
        <line
          fill="none"
          height={360}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={410}
          x={60}
          x1={60}
          x2={60}
          y={10}
          y1={10}
          y2={370}
        />
        <line
          fill="none"
          height={360}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={410}
          x={60}
          x1={470}
          x2={470}
          y={10}
          y1={10}
          y2={370}
        />
      </g>
    </g>
    <g
      className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
    >
      <line
        className="recharts-cartesian-axis-line"
        fill="none"
        height={30}
        stroke="#666"
        width={410}
        x={60}
        x1={60}
        x2={470}
        y={370}
        y1={370}
        y2={370}
      />
      <g
        className="recharts-cartesian-axis-ticks"
      >
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={30}
            stroke="#666"
            width={410}
            x={60}
            x1={60}
            x2={60}
            y={370}
            y1={376}
            y2={370}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={30}
            stroke="none"
            textAnchor="middle"
            width={410}
            x={60}
            y={378}
          >
            <tspan
              dy="0.71em"
              x={60}
            >
              5
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={30}
            stroke="#666"
            width={410}
            x={60}
            x1={470}
            x2={470}
            y={370}
            y1={376}
            y2={370}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={30}
            stroke="none"
            textAnchor="middle"
            width={410}
            x={470}
            y={378}
          >
            <tspan
              dy="0.71em"
              x={470}
            >
              5
            </tspan>
          </text>
        </g>
      </g>
    </g>
    <g
      className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
    >
      <line
        className="recharts-cartesian-axis-line"
        fill="none"
        height={360}
        stroke="#666"
        width={60}
        x={0}
        x1={60}
        x2={60}
        y={10}
        y1={10}
        y2={370}
      />
      <g
        className="recharts-cartesian-axis-ticks"
      >
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={360}
            stroke="#666"
            width={60}
            x={0}
            x1={54}
            x2={60}
            y={10}
            y1={370}
            y2={370}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={360}
            stroke="none"
            textAnchor="end"
            width={60}
            x={52}
            y={370}
          >
            <tspan
              dy="0.355em"
              x={52}
            >
              0
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={360}
            stroke="#666"
            width={60}
            x={0}
            x1={54}
            x2={60}
            y={10}
            y1={298}
            y2={298}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={360}
            stroke="none"
            textAnchor="end"
            width={60}
            x={52}
            y={298}
          >
            <tspan
              dy="0.355em"
              x={52}
            >
              1
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={360}
            stroke="#666"
            width={60}
            x={0}
            x1={54}
            x2={60}
            y={10}
            y1={226}
            y2={226}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={360}
            stroke="none"
            textAnchor="end"
            width={60}
            x={52}
            y={226}
          >
            <tspan
              dy="0.355em"
              x={52}
            >
              2
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={360}
            stroke="#666"
            width={60}
            x={0}
            x1={54}
            x2={60}
            y={10}
            y1={154}
            y2={154}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={360}
            stroke="none"
            textAnchor="end"
            width={60}
            x={52}
            y={154}
          >
            <tspan
              dy="0.355em"
              x={52}
            >
              3
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={360}
            stroke="#666"
            width={60}
            x={0}
            x1={54}
            x2={60}
            y={10}
            y1={81.99999999999999}
            y2={81.99999999999999}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={360}
            stroke="none"
            textAnchor="end"
            width={60}
            x={52}
            y={81.99999999999999}
          >
            <tspan
              dy="0.355em"
              x={52}
            >
              4
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={360}
            stroke="#666"
            width={60}
            x={0}
            x1={54}
            x2={60}
            y={10}
            y1={10}
            y2={10}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={360}
            stroke="none"
            textAnchor="end"
            width={60}
            x={52}
            y={10}
          >
            <tspan
              dy="0.355em"
              x={52}
            >
              5
            </tspan>
          </text>
        </g>
      </g>
    </g>
    <g
      className="recharts-layer recharts-area"
    >
      <g
        className="recharts-layer"
      >
        <defs>
          <clipPath
            id="animationClipPath-recharts-area-3"
          >
            <rect
              height={371}
              width={0}
              x={60}
              y={0}
            />
          </clipPath>
        </defs>
        <g
          className="recharts-layer"
          clipPath="url(#animationClipPath-recharts-area-3)"
        >
          <g
            className="recharts-layer"
            clipPath={null}
          >
            <path
              className="recharts-curve recharts-area-area"
              d="M60,226L470,226L470,370L60,370Z"
              fill="url(#colorUv)"
              fillOpacity={0.8}
              height={360}
              stroke="none"
              width={410}
            />
            <path
              className="recharts-curve recharts-area-curve"
              d="M60,226L470,226"
              fill="none"
              fillOpacity={0.8}
              height={360}
              stroke="#f15c41"
              width={410}
            />
          </g>
        </g>
      </g>
    </g>
    <defs>
      <linearGradient
        id="colorUv"
        x1="0"
        x2="0"
        y1="0"
        y2="1"
      >
        <stop
          offset="5%"
          stopColor="#e052a0"
          stopOpacity={1}
        />
        <stop
          fillOpacity={1}
          offset="95%"
          stopColor="#f15c41"
          stopOpacity={1}
        />
      </linearGradient>
    </defs>
  </svg>
  <div
    className="recharts-tooltip-wrapper"
    style={
      Object {
        "MozTransform": "translate(undefinedpx, undefinedpx)",
        "OTransform": "translate(undefinedpx, undefinedpx)",
        "WebkitTransform": "translate(undefinedpx, undefinedpx)",
        "msTransform": "translate(undefinedpx, undefinedpx)",
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "transform": "translate(undefinedpx, undefinedpx)",
        "visibility": "hidden",
      }
    }
  />
</div>
`;
